### PR TITLE
Features/guyber/filter resources emulator

### PR DESCRIPTION
**Pull request by bermanguy**
Filter resources in TreeView according to supported type combobox selection.
Also - enumerate resources before sending to detector.

image

image

### CODE REVIEW

File: src/SDK/MonitoringApplianceEmulator/ViewModels/SmartDetectorConfigurationControlViewModel.cs [hunk 1]
```diff
@@ -13,11 +13,13 @@ namespace Microsoft.Azure.Monitoring.SmartDetectors.MonitoringAppliancEmulator.V
     using System.Threading;
     using System.Threading.Tasks;
     using System.Windows;
+    using Management.ResourceManager.Fluent;
     using Microsoft.Azure.Monitoring.SmartDetectors;
     using Microsoft.Azure.Monitoring.SmartDetectors.Clients;
     using Microsoft.Azure.Monitoring.SmartDetectors.MonitoringAppliancEmulator.Controls;
     using Microsoft.Azure.Monitoring.SmartDetectors.MonitoringAppliancEmulator.Models;
     using Microsoft.Azure.Monitoring.SmartDetectors.Package;
+    using MonitoringApplianceEmulator;
```
Comment (EranGon) [inline src/SDK/MonitoringApplianceEmulator/ViewModels/SmartDetectorConfigurationControlViewModel.cs]: >MonitoringApplianceEmulator [](start = 10, length = 27)

full path

File: src/SDK/MonitoringApplianceEmulator/ViewModels/SmartDetectorConfigurationControlViewModel.cs [hunk 2]
```diff
@@ -281,6 +304,40 @@ public HierarchicalResource SelectedResource
             }
         }
 
+        /// <summary>
+        /// Gets the filtered resources.
+        /// </summary>
+        public ObservableCollection<HierarchicalResource> FilteredResources
```
Comment (EranGon) [inline src/SDK/MonitoringApplianceEmulator/ViewModels/SmartDetectorConfigurationControlViewModel.cs]: >FilteredResources [](start = 58, length = 17)

where is this used?

File: src/SDK/MonitoringApplianceEmulator/ViewModels/SmartDetectorConfigurationControlViewModel.cs [hunk 3]
```diff
@@ -391,6 +448,39 @@ public void SelectResource(object selectedResource)
             this.SelectedResource = (HierarchicalResource)selectedResource;
         }
 
+        /// <summary>
+        /// Indicates to filter.
```
Comment (EranGon) [inline src/SDK/MonitoringApplianceEmulator/ViewModels/SmartDetectorConfigurationControlViewModel.cs]: >Indicates to filter. [](start = 12, length = 20)

to filter what?

File: src/SDK/MonitoringApplianceEmulator/ViewModels/SmartDetectorConfigurationControlViewModel.cs [hunk 4]
```diff
@@ -391,6 +448,39 @@ public void SelectResource(object selectedResource)
...
+        /// </summary>
+        /// <param name="resource">the resource</param>
+        /// <returns>true or false</returns>
```
Comment (EranGon) [inline src/SDK/MonitoringApplianceEmulator/ViewModels/SmartDetectorConfigurationControlViewModel.cs]: >true or false [](start = 21, length = 13)

please be more descriptive

File: src/SDK/MonitoringApplianceEmulator/ViewModels/SmartDetectorConfigurationControlViewModel.cs [hunk 5]
```diff
@@ -391,6 +448,39 @@ public void SelectResource(object selectedResource)
...
+        public bool ShouldFilterResource(HierarchicalResource resource)
```
Comment (EranGon) [inline src/SDK/MonitoringApplianceEmulator/ViewModels/SmartDetectorConfigurationControlViewModel.cs]: >ShouldFilterResource [](start = 20, length = 20)

this indicates that true result means that the resource should be filtered - please rename

File: src/SDK/MonitoringApplianceEmulator/ViewModels/SmartDetectorConfigurationControlViewModel.cs [hunk 6]
```diff
@@ -391,6 +448,39 @@ public void SelectResource(object selectedResource)
...
+        {
+            if (this.SelectedResourceType == "All")
+            {
+                return true;
+            }
+
+            ResourceType selectedType;
+            Enum.TryParse(this.SelectedResourceType, out selectedType); // Should handle exception here?
```
Comment (EranGon) [inline src/SDK/MonitoringApplianceEmulator/ViewModels/SmartDetectorConfigurationControlViewModel.cs]: >// Should handle exception here? [](start = 72, length = 32)

no
so change to Parse

File: src/SDK/MonitoringApplianceEmulator/Controls/SmartDetectorConfigurationControl.xaml [hunk 7]
```diff
@@ -64,26 +64,73 @@
                             ItemsSource="{Binding Path=SupportedResourceTypes}" SelectedItem="{Binding Path=SelectedResourceType}" />
 
 
-                        <TextBlock Grid.Row="3" Grid.Column="0" Style="{StaticResource ControlText}" Text="Target resource" Margin="0,0,0,10" />
+                        <Grid Grid.Row="3" Grid.Column="0">
```
[the remark this answers is not in the document]
Comment (bermanguy) [reply src/SDK/MonitoringApplianceEmulator/Controls/SmartDetectorConfigurationControl.xaml]: added to PR link

---
In reply to: [181654760](https://github.com/liadtal/Smart-Alerts/pull/129#discussion_r181654760) [](ancestors = 181654760)

File: src/SDK/MonitoringApplianceEmulator/ViewModels/SmartDetectorConfigurationControlViewModel.cs (hunk 5 above)
Comment (bermanguy) [reply src/SDK/MonitoringApplianceEmulator/ViewModels/SmartDetectorConfigurationControlViewModel.cs]: I updated the documentation.. should rename still?


---
In reply to: [181682531](https://github.com/liadtal/Smart-Alerts/pull/129#discussion_r181682531) [](ancestors = 181682531)

Comment (EranGon) [reply src/SDK/MonitoringApplianceEmulator/ViewModels/SmartDetectorConfigurationControlViewModel.cs]: yes - filter means filter out, you return true if the resource should not be filtered

---
In reply to: [181689149](https://github.com/liadtal/Smart-Alerts/pull/129#discussion_r181689149) [](ancestors = 181689149,181682531)

Comment (EranGon) [reply src/SDK/MonitoringApplianceEmulator/ViewModels/SmartDetectorConfigurationControlViewModel.cs]: ShouldDisplayResource

---
In reply to: [181709134](https://github.com/liadtal/Smart-Alerts/pull/129#discussion_r181709134) [](ancestors = 181709134,181689149,181682531)

**Pull request merged**